File: Javascript/UploadLocal.js
const axios = require("axios");
const fs = require("fs");

const image = fs.readFileSync("YOUR_IMAGE.jpg", {
  encoding: "base64",
});

const dataset_name = "your-dataset"; // Set Dataset Name (Found in Dataset URL)

axios({
  method: "POST",
  url: "https://api.roboflow.com/dataset/" + dataset_name + "/upload",
  params: {
    api_key: "YOUR_KEY", // Enter your API Key Here
    name: "YOUR_IMAGE.jpg",
    split: "train",
  },
  data: image,
  headers: {
    "Content-Type": "application/x-www-form-urlencoded",
  },
})
  .then(function (response) {
    console.log(response.data);
  })
  .catch(function (error) {
    console.log(error.message);
  });

File: Ruby/UploadLocal.rb
require 'base64'
require 'httparty'

encoded = Base64.encode64(File.open("YOUR_IMAGE.jpg", "rb").read)
dataset_name = "your-dataset" # Set Dataset Name (Found in Dataset URL)
api_key = "" # Your API KEY Here

params = "?api_key=" + api_key + 
"&name=YOUR_IMAGE.jpg" + 
"&split=train"

response = HTTParty.post(
    "https://api.roboflow.com/dataset/" + dataset_name + "/upload" + params,
    body: encoded, 
    headers: {
    'Content-Type' => 'application/x-www-form-urlencoded',
    'charset' => 'utf-8'
  })

  puts response

 